### PR TITLE
docs: update Link component to v2 docs

### PR DESCRIPTION
**Pull request by tbusillo**
## Description 
- Updates the `Link` documentation to use the new MDX format

### CODE REVIEW

File: docs/components/example.css [hunk 1]
```diff
@@ -66,6 +66,10 @@ body.cauldron--theme-dark {
   position: relative;
 }
 
+.Component__example__react .Link--lowercase {
+  margin: 0 var(--space-smaller);
```
Comment (tbusillo) [inline docs/components/example.css:70]: Added this in so there would be space between the examples where buttons are adjacent to a link. It was looking crowded without this. 

CleanShot 2023-05-05 at 01 35 32@2x

File: packages/react/src/components/Button/index.tsx [hunk 2]
```diff
@@ -28,7 +30,8 @@ const Button = forwardRef<HTMLButtonElement, ButtonProps>(
         Link: variant === 'link',
         Tag: variant === 'tag',
         'Button--tag': variant === 'tag',
-        'Button--thin': thin
+        'Button--thin': thin,
+        'Link--lowercase': lowercase
```
Comment (dequejosie) [inline packages/react/src/components/Button/index.tsx:34]: `Button--lowercase`?

File: docs/pages/components/Link.mdx [hunk 3]
```diff
@@ -0,0 +1,88 @@
+---
+title: Link
+description: A link component with accessible styling.
```
Comment (dequejosie) [inline docs/pages/components/Link.mdx:3]: Do we need to define what `link` is like what we do with `button`?

File: docs/pages/components/Link.mdx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    {
+      name: 'lowercase',
+      type: 'boolean',
+      description:
+        'When `true`, adds style to make text contents all lowercase.'
+    },
+    {
+      name: 'buttonRef',
+      type: 'function',
+      description: 'Ref function for the button element'
```
Comment (dequejosie) [inline docs/pages/components/Link.mdx:81]: nit: Period at the end.

File: docs/pages/components/Link.mdx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+The default variant for the `Link` component.
+
+```jsx example
+<Link>I am a link!</Link>
+```
+
+### Button
+
+When you need a semantic link with the visual appearance of a button.
+
```
Comment (dequejosie) [inline docs/pages/components/Link.mdx:26]: nit: a primary button?

File: docs/pages/components/Link.mdx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+<ComponentProps
+  children={{
+    required: true
+  }}
+  props={[
+    {
+      name: 'variant',
+      type: ['button', 'button-secondary'],
+      defaultValue: 'button',
```
Comment (scurker) [inline docs/pages/components/Link.mdx:65]: This is incorrect, there's no default value for variant.

**Review by scurker — CHANGES_REQUESTED** [review]
One small change needed.

**Review by scurker — APPROVED** [review]

**Pull request merged**